packages/fetchai/skills/erc1155_deploy/behaviours.py

```diff
@@ -0,0 +1,235 @@
+        strategy = cast(Strategy, self.context.strategy)
+
+        if self.context.ledger_apis.has_fetchai:
+            fet_balance = self.context.ledger_apis.token_balance(
+                FETCHAI, cast(str, self.context.agent_addresses.get(FETCHAI))
+            )
+            if fet_balance > 0:
+                self.context.logger.info(
+                    "[{}]: starting balance on fetchai ledger={}.".format(
+                        self.context.agent_name, fet_balance
+                    )
+                )
+            else:
+                self.context.logger.warning(
+                    "[{}]: you have no starting balance on fetchai ledger!".format(
+                        self.context.agent_name
+                    )
+                )
+
+        if self.context.ledger_apis.has_ethereum:
+            eth_balance = self.context.ledger_apis.token_balance(
+                ETHEREUM, cast(str, self.context.agent_addresses.get(ETHEREUM))
+            )
+            if eth_balance > 0:
+                self.context.logger.info(
+                    "[{}]: starting balance on ethereum ledger={}.".format(
+                        self.context.agent_name, eth_balance
+                    )
+                )
+            else:
+                self.context.logger.warning(
+                    "[{}]: you have no starting balance on ethereum ledger!".format(
+                        self.context.agent_name
+                    )
+                )
+
+        self._register_service()
+
+        contract = self.context.contracts.erc1155
+        if self.contract_address is None:
+            self.context.logger.info("Loading details from json")
+            contract.set_instance(self.context.ledger_apis.apis.get("ethereum"))
+
+            dm_message_for_deploy = contract.get_deploy_transaction(
+                deployer_address=self.context.agent_address,
+                ledger_api=self.context.ledger_apis.apis.get("ethereum"),
+                skill_callback_id=self.context.skill_id,
+            )
+            self.context.decision_maker_message_queue.put_nowait(dm_message_for_deploy)
+        else:
+            self.context.logger.info("Setting the address of the deployed contract")
+            contract.set_instance_w_address(
+                ledger_api=self.context.ledger_apis.apis.get("ethereum"),
+                contract_address=self.contract_address,
+            )
+            create_items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+            contract.create_item_ids(token_type=strategy.ft, token_ids=create_items)
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+        contract = self.context.contracts.erc1155
+        strategy = cast(Strategy, self.context.strategy)
+        if not contract.is_deployed or self.contract_address is None:
+            if contract.is_deployed and not contract.is_items_created:
+                create_items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```

Comment:
Don't do these magic vars..everything should be configurable in config files and then there should be `DEFAULT` constants